bats/index.bats

```diff
@@ -2316,3 +2316,89 @@ SQL
     [ "$status" -eq "1" ]
     [[ "$output" =~ "UNIQUE" ]] || false
 }
+
+@test "index: Overwriting index auto-generated by foreign key" {
+    dolt sql <<SQL
+CREATE TABLE parent (
+  pk bigint PRIMARY KEY,
+  v1 bigint,
+  INDEX (v1)
+);
+CREATE TABLE child (
+  pk varchar(10) PRIMARY KEY,
+  parent_value bigint,
+  FOREIGN KEY (parent_value)
+    REFERENCES parent(v1)
+);
+CREATE TABLE child_idx (
+  pk varchar(10) PRIMARY KEY,
+  parent_value bigint,
+  INDEX (parent_value),
+  FOREIGN KEY (parent_value)
+    REFERENCES parent(v1)
+);
+CREATE TABLE child_unq (
+  pk varchar(10) PRIMARY KEY,
+  parent_value bigint,
+  FOREIGN KEY (parent_value)
+    REFERENCES parent(v1)
+);
+CREATE TABLE child_non_unq (
+  pk varchar(10) PRIMARY KEY,
+  parent_value bigint,
+  FOREIGN KEY (parent_value)
+    REFERENCES parent(v1)
+);
+INSERT INTO parent VALUES (1, 1), (2, 2), (3, 3), (4, NULL), (5, 5), (6, 6), (7, 7);
+INSERT INTO child VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', 3), ('5', 5);
+INSERT INTO child_idx VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', 3), ('5', 5);
+INSERT INTO child_unq VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', NULL), ('5', 5);
+INSERT INTO child_non_unq VALUES ('1', 1), ('2', NULL), ('3', 3), ('4', 3), ('5', 5);
+SQL
+    dolt sql -q "CREATE INDEX abc ON child (parent_value);"

```

Comment:
As part of this test, after creating the new key, you should do some inserts / deletes that are blocked by the foreign key constraint to verify that it still works. Also some inserts to verify that the unique constraint is being honored.